eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.sun.org.apache.xpath.internal.operations.Bool;
+import javafx.util.Pair;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.Stack;
+
+/**
+ * Check Style Rule: Good Logging Practice
+ *  <ol>
+ *      <li>Non-static Logger instance.</li>
+ *      <li>Logger variable name should be 'logger' in public class and not exist in an implementation package.</li>
+ *      <li>Should not use any external logger class, only use ClientLogger. No slf4j, log4j, or other logging imports are allowed.</li>
+ *      <li>Should refer to the containing class in its constructor:
+ *      public class A {
+ *          private ClientLogger logger = new ClientLogger(A.class);
+ *      }
+ *      </li>
+ *  </ol>
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String CLIENT_LOOGER_PATH = "com.azure.core.util.logging.ClientLogger";
+    private static final String CLIENT_LOGGER = "ClientLogger";
+    private static final String SLF4J = "org.slf4j";
+    private static final String LOG4J = "org.apache.logging.log4j";
```

Comment:
You should look for any other popular logging frameworks to blacklist those as well. You should even consider checking for calls to `System.out` and `System.err` for printing to the console.